server/routes/api.js

```diff
@@ -1073,129 +1089,359 @@ router.post('/capcodeImport', isLoggedIn, function (req, res, next) {
+          res.status(200);
+          res.json(row);
+        } else {
+          row = {
+            "username": "",
+            "password": "",
+            "givenname": "",
+            "surname": "",
+            "email": "",
+            "role": "user",
+            "status": "active"
+          };
+          res.status(200);
+          res.json(row);
+        }
+      })
+      .catch((err) => {
+        logger.main.error(err);
+        return next(err);
+      })
+  });
+
+  router.route('/userCheck/email/:id')
+  .get(isAdmin, function (req, res, next) {
+    var id = req.params.id;
+    db.from('users')
+      .select('*')
+      .where('email', id)
+      .then((row) => {
+        if (row.length > 0) {
+          row = row[0]
+          res.status(200);
+          res.json(row);
+        } else {
+          row = {
+            "username": "",
+            "password": "",
+            "givenname": "",
+            "surname": "",
+            "email": "",
+            "role": "user",
+            "status": "active"
+          };
+          res.status(200);
+          res.json(row);
+        }
+      })
+      .catch((err) => {
+        logger.main.error(err);
+        return next(err);
+      })
+  });
+
+router.route('/user/:id')
+  .get(isAdmin, function (req, res, next) {
+    var id = req.params.id;
+    var defaults = {
+      "username": "",
+      "password": "",
+      "givenname": "",
+      "surname": "",
+      "email": "",
+      "role": "user",
+      "status": "active"
+    };
+    if (id == 'new') {
+      res.status(200);
+      res.json(defaults);
+    } else {
+      db.from('users')
+        .select('*')
+        .where('id', id)
+        .then(function (row) {
+          if (row.length > 0) {
+            row = row[0]
+            res.status(200);
+            res.json(row);
+          } else {
+            res.status(200);
+            res.json(defaults);
+          }
+        })
+        .catch((err) => {
+          logger.main.error(err);
+          return next(err);
+        })
+    }
+  })
+  .post(isAdmin, function (req, res, next) {
+    var id = req.params.id || req.body.id || null;
+    if (id == 'deleteMultiple') {
+      // do delete multiple
+      var idList = req.body.deleteList || [0, 0];
+      if (!idList.some(isNaN)) {
+        logger.main.info('Deleting: ' + idList);
+        db.from('users')
+          .del()
+          .where('id', 'in', idList)
+          .then((result) => {
+            res.status(200).send({ 'status': 'ok' });
+
+          }).catch((err) => {
+            res.status(500).send(err);
+          })
       } else {
-        throw 'Error parasing CSV header'
+        res.status(500).send({ 'status': 'id list contained non-numbers' });
       }
-    })
-    .catch((err) => {
-      res.status(500).send(err)
-      logger.main.error(err)
-    })
-});
+    } else {
+      if (req.body.username && req.body.email && req.body.givenname) {
+        if (id == 'new') {
```

Comment:
Why this if there's a post on /user? Seems redundant

Reply:
because it was there on the code i stole from you somewhere else 🤣

Reply:
This route existed purely for API i think, the UI doesn't use it... I may look at removing it. If submitting via API there is no way to validate required fields.

Reply:
I was wrong, the post method on /user exists purely for the api!

Can be removed if we don't think api access to create users is required for api keys